### PR TITLE
Experiments Service: client side

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #897.

/cc @erwinmombay

### CODE REVIEW

**Comment by dvoytenko** [comment]
@erwinmombay getting an exception on the use of "cookie" here, which is expected. But once approved, how do I actually whitelist this case?

File: src/experiments.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+   * Whether the specified experiment is on or off.
+   * @param {string} experimentId
+   * @return {boolean}
+   */
+  isOn(experimentId) {
+    const cookieString = this.win.document.cookie;
+    if (!cookieString) {
+      return false;
+    }
+    const name = `exp-${encodeURIComponent(experimentId)}`;
```
Comment (cramforce) [inline src/experiments.js]: amp-exp-

File: src/experiments.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const cookies = cookieString.split(';');
```
Comment (cramforce) [inline src/experiments.js]: Probably better to factor this into cookie.js

We want to maintain the presubmit protection against cookie use, of course.

File: src/experiments.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    for (let i = 0; i < cookies.length; i++) {
+      const cookie = cookies[i].trim();
+      const eq = cookie.indexOf('=');
+      if (eq == -1) {
+        continue;
+      }
+      if (cookie.substring(0, eq).trim() == name) {
```
Comment (cramforce) [inline src/experiments.js]: Why not a format like:

`exp-amp=foo,bar,baz`

?

**Comment by cramforce** [comment]
One more thing: You need to whitelist the cookie to not be deleted automatically by our CDN.

File: src/experiments.js (hunk 1 above)
Comment (dvoytenko) [inline src/experiments.js]: Done

File: src/experiments.js (hunk 2 above)
Comment (dvoytenko) [inline src/experiments.js]: Done

File: src/experiments.js (hunk 3 above)
Comment (dvoytenko) [inline src/experiments.js]: Done

**Comment by dvoytenko** [comment]
@cramforce All done.
@erwinmombay For whitelists of cookie/getCookie

File: src/cookies.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * go through the privacy review. Before using this method please file a
+ * GitHub issue with "Privacy Review" label.
+ *
+ * Returns the cookie's value or `null`.
+ *
+ * @param {!Window} win
+ * @param {string} name
+ * @return {?string}
+ */
+export function getCookie(win, name) {
```
Comment (cramforce) [inline src/cookies.js]: Lets blacklist/whitelist getCookie

Comment (dvoytenko) [inline src/cookies.js]: Yeah. Already did :)

**Comment by cramforce** [comment]
LGTM

**Comment by erwinmombay** [comment]
LGTM

**Pull request merged**